### PR TITLE
feat: child logger

### PR DESCRIPTION
**Pull request by LucasdRossi**
closes [#125](https://github.com/Stand-With-Crypto/swc-internal/issues/125)

fixes 

## How has it been tested?

- [x] Locally
- [ ] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

### CODE REVIEW

**Review by olavoparno — APPROVED** [review]

File: src/inngest/functions/setPrimaryCryptoAddressOfUser/logic.ts [hunk 1]
```diff
@@ -13,11 +11,12 @@ const zodPrimaryCryptoAddressOfUserParameters = object({
 
 export async function setPrimaryCryptoAddressOfUser(
   parameters: z.infer<typeof zodPrimaryCryptoAddressOfUserParameters>,
+  logger?: Logger,
```
Comment (lucasrmp-cb) [inline src/inngest/functions/setPrimaryCryptoAddressOfUser/logic.ts:14]: Same here, if the idea is make `setPrimaryCryptoAddressOfUser` resilient we should just default this value to the return of `getLogger`
```suggestion
  logger: Logger = getLogger('setPrimaryCryptoAddressOfUser'),
```

File: src/inngest/functions/sms/utils/enqueueMessages.ts [hunk 2]
```diff
@@ -35,7 +33,11 @@ export interface EnqueueMessagePayload {
   messages: PayloadMessage[]
 }
 
-export async function enqueueMessages(payload: EnqueueMessagePayload[], attempt = 0) {
+export async function enqueueMessages(
+  payload: EnqueueMessagePayload[],
+  logger?: Logger,
```
Comment (lucasrmp-cb) [inline src/inngest/functions/sms/utils/enqueueMessages.ts:38]: Same here, and let's merge `logger` and `attempt` in a `options` object, the API is easier to read that way

**Review by lucasrmp-cb — CHANGES_REQUESTED** [review]

File: src/inngest/functions/backfillCongressionalDistrictCronJob/index.ts [hunk 3]
```diff
@@ -94,6 +93,7 @@ async function backfillUsCongressionalDistricts(
     Address,
     'id' | 'formattedDescription' | 'countryCode'
   >[],
+  logger?: Logger,
```
[the remark this answers is not in the document]
Comment (LucasdRossi) [reply src/inngest/functions/backfillCongressionalDistrictCronJob/index.ts:96]: I thought about not linking the logic of the functions to a specific type in Inngest. There are some functions that separate the logic from the initialization and execution of Inngest:
[src/inngest/functions/backfillNFT
](https://github.com/Stand-With-Crypto/swc-web/tree/main/src/inngest/functions/backfillNFT)
[src/inngest/functions/cleanupPostalCodes ](https://github.com/Stand-With-Crypto/swc-web/tree/main/src/inngest/functions/cleanupPostalCodes)

**Review by olavoparno — DISMISSED** [review]

File: src/inngest/functions/backfillCongressionalDistrictCronJob/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lucasrmp-cb) [reply src/inngest/functions/backfillCongressionalDistrictCronJob/index.ts]: You have a few options to suppress or even test console calls in jest, I think we can go towards something like this instead.
image
Maybe we wanna mock logs based on and env or something like that. It serves as a test debug sometimes, I dont think we should change implementation thinking of tests but the other way around

[the remark this answers is not in the document]
Comment (lucasrmp-cb) [reply src/inngest/functions/backfillCongressionalDistrictCronJob/index.ts]: And about not creating inngest dependencies. I get that and agree to some extend, but I'm ok with creating inngest dependencies inside the `inngest` folder

File: src/inngest/functions/cleanupPostalCodes/logic.ts [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import { prismaClient } from '@/utils/server/prismaClient'
-import { Logger } from '@/utils/shared/logger'
+import { getLogger } from '@/utils/shared/logger'
 
-export async function cleanPostalCodes(persist: boolean, logger?: Logger) {
+const defaultLogger = getLogger('cleanPostalCodes')
+
+export async function cleanPostalCodes(persist: boolean, logger = defaultLogger) {
```
Comment (lucasrmp-cb) [inline src/inngest/functions/cleanupPostalCodes/logic.ts:6]: its still a np of mine (and you don't need to change it), but I don't like the signature call of `cleanPostalCodes(true, logger)`, I'd much prefer `cleanPostalCodes({ persist: true, logger })`, this way I know what `true` means

File: src/inngest/functions/setPrimaryCryptoAddressOfUser/logic.ts [hunk 5]
```diff
@@ -36,7 +38,7 @@ export async function setPrimaryCryptoAddressOfUser(
     throw new Error('a user with the same primaryUserCryptoAddressId already exist')
   }
   if (!persist) {
-    logger?.info('Dry run, exiting')
+    logger.info('Dry run, exiting')
```
Comment (lucasrmp-cb) [inline src/inngest/functions/setPrimaryCryptoAddressOfUser/logic.ts:41]: I know this is not your task, but I think we're missing a return here. Big problem so if you can check and fix

**Review by lucasrmp-cb — CHANGES_REQUESTED** [review]

**Review by olavoparno — APPROVED** [review]

**Review by lucasrmp-cb — APPROVED** [review]

**Pull request merged**